src/mscorlib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -181,6 +181,8 @@ internal static unsafe int LastIndexOfOrdinalCore(string source, string value, i
         private static unsafe int CompareStringOrdinalIgnoreCase(char* string1, int count1, char* string2, int count2)
         {
             Debug.Assert(!GlobalizationMode.Invariant);
+            Debug.Assert(string1 != null);
```

Comment:
The assert in this file are unnecessary. The Unix globalizations API handle NULL with zero-sized buffer just fine.

Reply:
(You can keep them if you would like though.)